lib/imapSimple.js

```diff
@@ -225,9 +261,49 @@ function connect(options, callback) {
     });
 }
 
+/**
+ * Returns a list of the parts.
+ * A part is either a message or an attachment.
+ * Parts are usually nested and numbered things like 1, 1.1, 1.2, ...
+ * Code taken from http://stackoverflow.com/questions/25247207/how-to-read-and-save-attachments-using-node-imap
+ */
+function findParts(struct, parts) {
```

Comment:
okay, yeah, this would be pretty useful for flattening a message's `attributes.struct`. Maybe name it `getParts` since it's not really a search but more a flattening/retrieval?